src/provisioner/validators.py

```diff
@@ -148,30 +147,61 @@ def _check_number_of_nodes(self):
     def _check_external_services(self):
         """" Validate no of external services in cortx config """
 
-        sol_conf_ext_services = self._get_value_from_solution_config(
+        cortx_config_ext_services = self._get_config(
             ConfigValidator._key_ext_service)
         conf_stor_ext_services = self._get_value_from_conf_store(
             ConfigValidator._key_ext_service)
 
-        if len(conf_stor_ext_services) != len(sol_conf_ext_services):
+        if len(conf_stor_ext_services) != len(cortx_config_ext_services):
             raise CortxProvisionerError(errno.EINVAL,
                 'External services define in conf_store is not same as '
-                f'solution_config. External services in {self.solution_conf_url}'
-                f' is {len(sol_conf_ext_services)} and in {self.cortx_conf_url}'
+                f'cortx config. External services in {self.solution_conf_url}'
+                f' is {len(cortx_config_ext_services)} and in {self.cortx_conf_url}'
                 f' is {len(conf_stor_ext_services)}')
 
         # Check endpoints
         for service in conf_stor_ext_services.keys():
             if (conf_stor_ext_services[service]['endpoints'] !=
-                    sol_conf_ext_services[service]['endpoints']):
+                    cortx_config_ext_services[service]['endpoints']):
                 raise CortxProvisionerError(errno.EINVAL,
                     f'{service} endpoints define in conf_store is not same as '
-                    'solution_config.')
+                    'cortx config.')
         return 0
 
-    def _get_value_from_solution_config(self, key):
-        """ Read config value for key from solution_config """
+    def _validate_components(self):
+        """Verify components defined in cluster.yaml is supported in constant file."""
+        node_types = self._get_config('cluster>node_types')
+        for node_type in node_types:
+            components_list = node_type['components']
+            for component in components_list:
+                # check component_name exist in Utils Const cls.
+                component_name = component['name']
+                if component_name not in Const._value2member_map_:
+                    raise CortxProvisionerError(errno.EINVAL,
+                        f'{component_name} is not present in constant file.')
+                if component.get('services') is not None:
+                    services = component.get('services')
+                    self._validate_services(services, component_name)
+
+    def _validate_services(self, services, component_name):
+        """Verify services defined in cluster.yaml is supported in constant file."""
+        for service in services:
+            # check if service name is define in utils const.py file.
+            if service not in Const._value2member_map_:
+                raise CortxProvisionerError(errno.EINVAL,
+                    f'{service} is not present in constant file.')
```

Comment:
Make error generic something like:
Value not supported...

Reply:
Done